### ISSUE TITLE
Returning user info confirmation flow

### ISSUE THREAD

**Issue by hellafitz** [Iridescent-CM/technovation-app#2583]
ticket to make a new location confirmation or key profile info confirmation flow (there may be a hack vs. a long term flow)

general concept: show returning users a simplified version of the registration screens that allows them to confirm the information we have for them, or update/edit it as desired. 

reference doc: Designing Season Rollover, https://docs.google.com/document/d/1oBVG_RMi6SVQ8KlznqmdRT_FuNCgRCkBy8maM4tXeWA/edit#heading=h.7wh191bk4ijs

**Comment by hellafitz** [comment]
Mockups are specifically for direction 2, below. 

potential directions:
1) hack approach: everyone keeps their account but we delete key info which the platform forces them to fill in again because it needs to be there (may not work for everything we want people to reconfirm)
2) yes/and approach: we build on the existing interruptions screen patter, similar to Data & Agreements or Location interrupts, and this new interrupt screen doesn't let returning users go anywhere until they've done it (and we'd make a way to selectively apply this to returning users, of specific types, only)
3) remodel approach: redesign season registration (make them opt-in rather than the platform assuming that they're here so they're signing up). This is related to the question of what it means to be on the platform but not registered for a season (like if you show up late in the season after submissions have closed).

Additional context: the platform would need work to make it recognize some version of "they're here but they haven't confirmed everything they need to, so don't register them yet." The platform very aggressively registers people to the current season (because it deletes most old data then makes people re-enter everything).

**Comment by hellafitz** [comment]
Notes:
- for returning users, the Data Agreement screen is the same (we think) but the flow is different. 
- when gathering screenshots, try different user types as there may be variations

**Comment by rsgonzal** [comment]
1. logging in as: roxana+nds33@technovation.org
2. Brought me to the dashboard
3. sent an email to my parent email on file
4. sent an email welcoming me to the season

does not ask me for data terms agreement

**Comment by rsgonzal** [comment]
returning who will now be a beginner
1. logging in as roxana+inviteflow@technovation.org
2. looks like i have to fill out my parent information, doesn't give me an option for paper consent (amazing!)
![Screen Shot 2021-11-24 at 7.51.12 PM.png](https://images.zenhubusercontent.com/60527169cde2772242d0de95/b81e25fa-ecab-4b4d-b5cc-129af5fc74f9)
3. i am able to download after actually
![Screen Shot 2021-11-24 at 7.54.00 PM.png](https://images.zenhubusercontent.com/60527169cde2772242d0de95/8e0763c9-d3bc-4e2b-976f-2b192ed94440)
4. site puts me in beginner division
![Screen Shot 2021-11-24 at 7.56.35 PM.png](https://images.zenhubusercontent.com/60527169cde2772242d0de95/80e6021a-73d6-400d-8ce1-bc86fb042a9b)

was not asked to change email. noticing parent name not filled out from last year (mustve been paper consent)

**Comment by rsgonzal** [comment]
i guess the first screenshot above is always how it's set up. interesting!

**Comment by shaunxp20** [comment]
> does not ask me for data terms agreement

When we run the `reset_for_season` rake task, it will reset the data terms agreement, line no. 13 (`terms_agreed_at`, is referring to the data terms).

https://github.com/Iridescent-CM/technovation-app/blob/96404ee861351acf39714176566a16007a4133b6/lib/tasks/reset_for_season.rake#L11-L19

I don't see anything in there explicitly mentioning the parental consent, somehow it worked as expected w/ this account, which means something is happening elsewhere (outside of this rake task) to make that happen, or it was just a coincidence for this account. 🤔 🧐 🤨 😄

**Comment by rsgonzal** [comment]
you mean to retake the parent consent? that happens on its own for season rollover

**Comment by shaunxp20** [comment]
> you mean to retake the parent consent? that happens on its own for season rollover

Yeah, I didn't see that in the rake task, but good to know it happens on its own, I think we're good! 👍
